sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/aio/_schema_registry_avro_encoder_async.py

```diff
@@ -0,0 +1,269 @@
+
+    def _convert_preamble_format(self, data, content_type):  # pylint: disable=no-self-use
+        record_format_identifier = b"\0\0\0\0"
+        if data[0:RECORD_FORMAT_IDENTIFIER_LENGTH] == record_format_identifier:
+            schema_id = data[
+                SCHEMA_ID_START_INDEX : (SCHEMA_ID_START_INDEX + SCHEMA_ID_LENGTH)
+            ].decode("utf-8")
+            content_type = f"{AVRO_MIME_TYPE}+{schema_id}"
+            data = data[DATA_START_INDEX:]
+
+        return data, content_type
+
+    async def decode(
+        self,
+        *,
+        message: Optional[MessageType] = None,
+        data: Optional[bytes] = None,
+        content_type: Optional[str] = None,
+        **kwargs,
+    ) -> Dict[str, Any]:
+        """
+        Decode bytes data using schema ID in the content type field. One of the following is required:
+            1) `message`
+            2) `data` and `content_type`
+        Data must follow format of associated Avro RecordSchema:
+        https://avro.apache.org/docs/1.10.0/gettingstartedpython.html#Defining+a+schema
+
+        :param bytes value: The bytes data needs to be decoded.
+        :keyword message: The message object which holds the data to be decoded and content type
+         containing the schema ID. If not set, both `data` and `content_type` must be set.
+        :paramtype message: MessageType or None
+        :keyword data: The data to be decoded. Must be set with `content_type`. If not set, `message` must be set.
+        :paramtype data: bytes or None
+        :keyword content_type: The content type containing the schema ID to be used for decoding.
+         Must be set with `data`. If not set, `message` must be set.
+        :paramtype content_type: str or None
+        :rtype: Dict[str, Any]
+        :raises ~azure.schemaregistry.encoder.avroencoder.exceptions.SchemaParseError:
+            Indicates an issue with parsing schema.
+        :raises ~azure.schemaregistry.encoder.avroencoder.exceptions.SchemaDecodeError:
+            Indicates an issue with decoding value.
+        """
+
+        if message:
+            try:
+                data = message.__data__()
+                content_type = message.__content_type__()
+            except AttributeError:
+                SchemaDecodeError(
+                    f"""The data model {str(message)} does not support MessageType protocol.
+                        If using an Azure SDK model class, please check the README.md for the full list
+                        of supported Azure SDK models and their corresponding versions."""
+                ).raise_with_traceback()
+        if not data:
+            raise ValueError("'data' value cannot be None.")
+
+        # include in first preview for back compatibility
+        data, content_type = self._convert_preamble_format(data, content_type)
+        if not content_type:
+            raise ValueError("'content' type cannot be None.")
```

Comment:
nit: should be this?
```suggestion
            raise ValueError("'content_type' cannot be None.")
```